iPetCare/Application/Services/PetService.cs

```diff
@@ -0,0 +1,218 @@
+
+                if (!await Context.VetPets.AnyAsync(vp => vp.PetId == pet.Id && vp.VetId == vet.Id))
+                    return new ServiceResponse<PetsGetPetDtoResponse>(HttpStatusCode.Forbidden);
+
+                var petToReturn = Mapper.Map<PetsGetPetDtoResponse>(pet);
+                return new ServiceResponse<PetsGetPetDtoResponse>(HttpStatusCode.OK, petToReturn);
+            }
+
+            return new ServiceResponse<PetsGetPetDtoResponse>(HttpStatusCode.Forbidden);
+        }
+
+        public async Task<ServiceResponse<PetsCreatePetDtoResponse>> CreatePetAsync(PetsCreatePetDtoRequest dto)
+        {
+            if(dto.Gender == null)
+                return new ServiceResponse<PetsCreatePetDtoResponse>(HttpStatusCode.BadRequest, "Gender must be specified");
+
+            var username = CurrentlyLoggedUserName;
+            var user = await Context.Users.SingleOrDefaultAsync(u => u.UserName == username);
+            if (user == null)
+                return new ServiceResponse<PetsCreatePetDtoResponse>(HttpStatusCode.BadRequest, "User not found");
+
+            var owner = await Context.Owners.SingleOrDefaultAsync(o => o.UserId == user.Id);
+            if (owner == null)
+                return new ServiceResponse<PetsCreatePetDtoResponse>(HttpStatusCode.BadRequest, "Owner not found");
+
+            if(await Context.Pets.AnyAsync(p => p.Id == dto.Id))
+                return new ServiceResponse<PetsCreatePetDtoResponse>(HttpStatusCode.BadRequest, "Pet with given id already exists, try again with other id");
+
+            Pet pet = Mapper.Map<Pet>(dto);
+
+            var race = await Context.Races.SingleOrDefaultAsync(r => r.Id == dto.RaceId);
+            if (race == null)
+                return new ServiceResponse<PetsCreatePetDtoResponse>(HttpStatusCode.BadRequest, "Invalid race");
+
+            pet.Race = race;
+
+            // if not given from the front
+            if (pet.Id == Guid.Empty)
+                pet.Id = Guid.NewGuid();
+
+            Context.Pets.Add(pet);
+            Context.OwnerPets.Add(new OwnerPet
+            {
+                Pet = pet,
+                Owner = owner
+            });
+
+            if (await Context.SaveChangesAsync() <= 0)
+                return new ServiceResponse<PetsCreatePetDtoResponse>(HttpStatusCode.BadRequest,
+                    "An error occured while creating pet");
+
+            var petToReturn = Mapper.Map<PetsCreatePetDtoResponse>(pet);
+            return new ServiceResponse<PetsCreatePetDtoResponse>(HttpStatusCode.OK, petToReturn);
+        }
+
+        public async Task<ServiceResponse<PetsUpdatePetDtoResponse>> UpdatePetAsync(Guid petId,
+            PetsUpdatePetDtoRequest dto)
+        {
+            if (petId == Guid.Empty)
+                return new ServiceResponse<PetsUpdatePetDtoResponse>(HttpStatusCode.BadRequest,
+                    "Pet id must be specified");
+
+            if (petId != dto.Id)
```

Comment:
Czy na pewno robimy tak że i w dto i w url jest id? Czy to jest potrzebne?

Reply:
Usunięte